src/server/route/routes/user.ts

```diff
@@ -0,0 +1,110 @@
+import { router } from '../router';
+import { Request, Response } from 'express';
+import { db } from '../../dbConfigs';
+import { Login, Registration } from '../../../../types';
+import bcrypt from 'bcrypt';
+import * as jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+dotenv.config();
+
+router.route('/user/register').post(async (req: Request, res: Response) => {
+    const user: Registration = req.body;
+
+    if (!user || !user.username || !user.password || !user.email) {
+        res.status(403).json({ message: 'Missing parameters' });
+        return;
+    }
+
+    const saltRounds = 10;
+    const hash = await bcrypt.hash(user.password, saltRounds);
+    const q = await db.query(
+        'SELECT username FROM users WHERE username=$1 OR email=$2',
+        [user.username, user.email]
+    );
+    if (q.rowCount == 0) {
+        await db.query(
```

Comment:
It might be neater to use INSERT ... ON CONFLICT DO NOTHING and checking the returned row count. Or not.